Config module tests

SatyendraBanjare opened the pull request:
Added Config and Buf Module Tests. :smile:

SatyendraBanjare committed:
Tests : Added Buf Module Tests

source : Modified Buf methods, updated luacov

SatyendraBanjare committed:
Source : Added new Config module bindings & Helper methods

SatyendraBanjare committed:
Source : update Installation instructions

SatyendraBanjare committed:
Tests : Modified repo fixture methods

SatyendraBanjare committed:
Tests : Added Config Module Tests

Also Updated luacov, and buf tests

SatyendraBanjare committed:
Tests : Added Tree Module Tests

SatyendraBanjare committed:
source : Removed error with buf_free method

SatyendraBanjare committed:
Tests: Added Signature Module Tests

SatyendraBanjare committed:
Tests: Added Object Module Tests

SatyendraBanjare committed:
Tests: Added Tests for Tag Module

SatyendraBanjare committed:
Source : Added Cred helpers

SatyendraBanjare committed:
Tests: Added Tests for Cred Module

SatyendraBanjare committed:
source : added some functions to repository

SatyendraBanjare committed:
Tests : Added Repository Module Tests

SatyendraBanjare committed:
Source : Added Index module helper functions

Also registered functions in mainwrapper and updated installation instructions

SatyendraBanjare committed:
Tests : Added Index Module Tests

SatyendraBanjare committed:
Tests: Added Reference module tests

Fixed a minor error in setting table name in a function in reference  module

SatyendraBanjare committed:
refactor[source(libgit)] : check arg count, removed helper function

Modified some functions to make luagit2 more lua_way, accessing some options in functions using string made available

SatyendraBanjare committed:
refactor[source(main_wrapper)] changed names of all functions

 also updated installation instructions

SatyendraBanjare committed:
refactor[tests(libgit)] : improved styling to match new functions

SatyendraBanjare committed:
Refactor: modified oid module source and thus its tests

SatyendraBanjare committed:
refactor : modified tree module source and thus its tests

SatyendraBanjare committed:
Refactor : Modified signature module source + tests

Also slightly modified index module helper function

SatyendraBanjare committed:
Refactor : modified tag module source code and tests

SatyendraBanjare committed:
refactor : index module source + tests

SatyendraBanjare committed:
source (annotated) : added bindings

SatyendraBanjare committed:
updated installation instructions

SatyendraBanjare committed:
tests : added annotated module tests

Etiene commented on src/luagit2/config/lua_config_help.c:
```diff
@@ -0,0 +1,100 @@
+// 	 * specific config file available that actually is loaded)
+// 	 */
+// 	GIT_CONFIG_HIGHEST_LEVEL = -1,
+// } git_config_level_t;
```
Why was this commented out?

SatyendraBanjare replied:
That is my mistake, I used to refer it while creating bindings. It has absolutely nothing to do with any of code. I will remove it. Thankyou

Etiene commented on tests/blob/test_blob_spec.lua:
```diff
@@ -1,7 +1,7 @@
 
 local fixer = require("Fixtures.fix_repo")
-local lfs = require("lfs")
+
```
why is this removed here?

SatyendraBanjare replied:
There was no use of lfs module to check the path in this test. Therefore I removed it.

Etiene replied:
ok!

Etiene commented on tests/buf/test_buf_spec.lua:
```diff
@@ -0,0 +1,35 @@
+	--
+	it("Tests Buf Free Method",function()
+		local buffer = lib.luagit2_buf_set_str("Hello")
+		--lib.luagit2_buf_free(buffer)
```
This test isnt doing anything! Please adjust :)

SatyendraBanjare replied:
I have fixed it in a later commit.

Etiene replied:
can you post the link here so I have a look?

SatyendraBanjare replied:
https://github.com/SatyendraBanjare/luagit2/blob/0e1049311557a8895b9fa1686b14b22cc96df14a/tests/buf/test_buf_spec.lua#L14

Etiene commented on tests/config/test_config_spec.lua:
```diff
@@ -0,0 +1,363 @@
+
+		local global_email_handle = io.popen("git config --global --get user.email ")
+		global_email = global_email_handle:read("*a")
+		global_email_handle:close()
```
It looks like there is a lot of repetition here. One way to improve it write a local function at the top of test file, just before the first describe that encapsulates opening a command, reading and closing it, then you just pass the command string here :)

SatyendraBanjare replied:
ok thankyou. I will try to do so.

Etiene commented on tests/config/test_config_spec.lua:
```diff
@@ -0,0 +1,363 @@
+
+	-- 	assert.are.equal(name .. '\n', sys_name)
+	-- 	assert.are.equal(email .. '\n',sys_email)
+	-- end)
```
Why is this test commented out?

SatyendraBanjare replied:
this function `luagit2_config_open_system()` was giving errors. so I commented that part out.

SatyendraBanjare replied:
I will try to fix it.

Etiene replied:
ok! :)  remember to add the fixes on this PR and not a different PR :)

SatyendraBanjare committed:
Fixed some errors, modified tests & removed unnecessary comments

SatyendraBanjare commented:
@Etiene  mam, I have fixed those few small errors. I have removed those unnecessary comments. Also I realized there is no extra function to open system level config, instead we can do that using `open_config_level()` and then passing system as the option.

Etiene commented on tests/config/test_config_spec.lua:
```diff
@@ -0,0 +1,345 @@
+local lfs = require("lfs")
+
+local function get_config_details(option_string)
+	name_handle = io.popen("git config --get --" .. option_string .." user.name ")
```
global variable here! are you using luacheck?

SatyendraBanjare committed:
tests(config): fixed global declaration error

SatyendraBanjare commented:
@Etiene  mam, I re installed my editor, forgot to install linter. Fixed the error now. :smile:

Etiene reviewed (approved):
👍
